be/src/storage/lake/pk_index_loader.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "storage/lake/pk_index_loader.h"
+
+#include <fmt/format.h>
+
+#include <chrono>
+
+#include "common/constexpr.h"
+#include "exec/workgroup/scan_executor.h"
+#include "exec/workgroup/work_group.h"
+#include "storage/chunk_helper.h"
+#include "storage/lake/tablet_manager.h"
+#include "storage/lake/update_manager.h"
+#include "storage/olap_common.h"
+#include "storage/rowset/segment_options.h"
+#include "util/defer_op.h"
+
+namespace starrocks::lake {
+
+class PkSegmentScanTask {
+public:
+    PkSegmentScanTask(TabletManager* tablet_mgr, const TabletMetadataPtr& metadata, uint32_t rowset_id, uint32_t seg_id,
+                      const std::string& seg_name, const Schema& schema, int64_t version,
+                      const MetaFileBuilder* builder)
+            : _tablet_mgr(tablet_mgr),
+              _metadata(std::move(metadata)),
```

Comment:
```suggestion
              _metadata(metadata),
```
The `std::move` here is totally useless.